oeps/best-practices/oep-0049-django-app-patterns.rst

```diff
@@ -267,7 +267,31 @@ Do note that even if you expose your tasks through ``api.py`` to be used by othe
 still be configured with the ``tasks`` import name, as the celery identifier for your task (as set by the celery
 decorator) is based off the original file.
 
+docs/decisions/0001-purpose-of-this-repo.rst or 0001-purpose-of-this-app.rst
```

Comment:
Actually that's an interesting point. If a new repo has only one new app, do we want both a purpose of this app and a purpose of this repo?

Reply:
My gut reaction is that requring separate ADRs for the repo *and* the app sounds like too much paperwork. However, requiring ADRs for apps created in exisitings repo does sound fair to me.

As a conpromise, we could recommend that the initial app(s) have stub ADRs, each of which point to the repo-level ADR. The repo cookiecutter could be updated to provide that.

Reply:
Yeah, I'd lost track of this being for apps in general, which might be in their own repos. If it's a single-app repo, I think there should just be one docs/ dir, either at the repo root or inside the app. Top-level sounds more convenient and easy to notice. The one reason I can think of to put it inside the app is in case the code moves into another repo (maybe absorbed into an IDA?), since having it at the repo root might risk it being lost. But I don't know how plausible that is.

Reply:
I think @kdmccormick 's suggestion of the stub ADR pointing to the main ADR addresses most of your concerns @timmc-edx . The documentation is effectively centralized, but if someone moves they app folder they should notice that something is up when their link breaks.

Reply:
I like that proposal as well.

Reply:
A stub makes to me as well, yeah.

Reply:
For what it's worth, I wrote that proposal thinking about apps in an IDA, not single-app packages.

I would agree with @timmc-edx's earlier point that a single ADR in the root dir of the a single-app package is probably good enough. But if that's not that the consensus that's been reached here, that's fine with me too.

Reply:
Updated with the stub-ADR-in-the-app version. Even I find the phrasing I used confusing so if anyone has ideas for ways to express this more clearly I'm very much open to them.

Reply:
[inform] The updated text is in https://github.com/openedx/open-edx-proposals/pull/455/files#r1122281928, but the PR hasn't been updated yet.

Reply:
PR has been updated